tests/kafkatest/services/verifiable_producer.py

```diff
@@ -56,20 +66,25 @@ def __init__(self, context, num_nodes, kafka, topic, max_messages=-1, throughput
         self.not_acked_values = []
         self.prop_file = ""
 
-
     def _worker(self, idx, node):
         node.account.ssh("mkdir -p %s" % VerifiableProducer.PERSISTENT_ROOT, allow_fail=False)
 
         # Create and upload log properties
         self.security_config = self.kafka.security_config.client_config(self.prop_file)
-        self.prop_file += str(self.security_config)
+        producer_prop_file = str(self.security_config)
         log_config = self.render('tools_log4j.properties', log_file=VerifiableProducer.LOG_FILE)
         node.account.create_file(VerifiableProducer.LOG4J_CONFIG, log_config)
 
         # Create and upload config file
+        if self.compression_types is not None:
```

Comment:
This will require pretty big change/rewrite to ProduceConsumeValidateTest. It currently assumes one VerifiableProducer. Let me just write a comment above this line.